src/accelerate/utils/dataclasses.py

```diff
@@ -173,32 +173,84 @@ class InitProcessGroupKwargs(KwargsHandler):
+
+    ```python
+    kwargs = FP8RecipeKwargs(backend="msamp", optimization_level="02")
+    ```
+
+    Args:
+        backend (`str`, *optional*, defaults to "msamp"):
+            Which FP8 engine to use. Must be one of `"msamp"` (MS-AMP) or `"te"` (TransformerEngine).
+        margin (`int`, *optional*, default to 0):
+            The margin to use for the gradient scaling.
+        interval (`int`, *optional*, default to 1):
+            The interval to use for how often the scaling factor is recomputed.
+        fp8_format (`str`, *optional*, default to "E4M3"):
+            The format to use for the FP8 recipe. Must be one of `E4M3` or `HYBRID`.
+        amax_history_len (`int`, *optional*, default to 1024):
+            The length of the history to use for the scaling factor computation
+        amax_compute_algo (`str`, *optional*, default to "most_recent"):
+            The algorithm to use for the scaling factor computation. Must be one of `max` or `most_recent`.
+        override_linear_precision (`tuple` of three `bool`, *optional*, default to `(False, False, False)`):
+            Whether or not to execute `fprop`, `dgrad`, and `wgrad` GEMMS in higher precision.
+        optimization_level (`str`), one of `O1`, `O2`. (default is `O2`):
+            What level of 8-bit collective communication should be used with MS-AMP. In general:
+                * O1: Weight gradients and `all_reduce` communications are done in fp8, reducing GPU
+                    memory usage and communication bandwidth
+                * O2: First-order optimizer states are in 8-bit, and second order states are in FP16.
+                    Only available when using Adam or AdamW. This maintains accuracy and can potentially save the
+                    highest memory.
+                * 03: Specifically for DeepSpeed, implements capabilities so weights and master weights of models
+                    are stored in FP8. If `fp8` is selected and deepspeed is enabled, will be used by default. (Not
+                    available currently).
     """
 
+    backend: Literal["msamp", "te"] = "msamp"
+    opt_level: Literal["O1", "O2"] = "O2"
     margin: int = 0
     interval: int = 1
-    fp8_format: str = "E4M3"
+    fp8_format: Literal["E4M3", "HYBRID"] = "E4M3"
     amax_history_len: int = 1
-    amax_compute_algo: str = "most_recent"
+    amax_compute_algo: Literal["max", "most_recent"] = "most_recent"
     override_linear_precision: Tuple[bool, bool, bool] = (False, False, False)
 
     def __post_init__(self):
-        self.fp8_format = self.fp8_format.upper()
-        if self.fp8_format not in ["E4M3", "HYBRID"]:
-            raise ValueError("`fp8_format` must be 'E4M3' or 'HYBRID'.")
-        if self.amax_compute_algo not in ["max", "most_recent"]:
-            raise ValueError("`amax_compute_algo` must be 'max' or 'most_recent'")
+        self.backend = self.backend.upper()
+        if self.backend not in ["MSAMP", "TE"]:
```

Comment:
Not a huge issue, just a tip: You can get the values from the `Literal`, which can help avoid duplication:

```python
from typing import Literal, get_args
FP8_Opt_Level = Literal["O1", "O2"]

...
    opt_level: FP8_Opt_Level = "O2"
...
            if self.opt_level not in get_args(FP8_Opt_Level):
...
```

Reply:
There are other cases where this could be good in the library to do, I'll change it here then a follow up to help other areas.